File: backend_services/insight-runtime-go/pkg/clubregistry/clubregistry_test.go
package clubregistry

import "testing"

func TestCanonicalResolution(t *testing.T) {
	r := Default()
	cases := []struct {
		query string
		want  string
	}{
		{"Manchester City", "manchester_city"},
		{"Man City", "manchester_city"},
		{"Manchester City FC", "manchester_city"},
		{"manchester_city", "manchester_city"},
		{"MCI", "manchester_city"},
		{"Flamengo", "flamengo"},
		{"CR Flamengo", "flamengo"},
		{"Clube de Regatas do Flamengo", "flamengo"},
		{"Atlético Mineiro", "atletico_mineiro"},
		{"Atlético-MG", "atletico_mineiro"},
		{"Barça", "barcelona"},
		{"FC Bayern München", "bayern_munich"},
	}
	for _, c := range cases {
		got, ok := r.Resolve(c.query)
		if !ok || got != c.want {
			t.Errorf("Resolve(%q) = %q,%v; want %q", c.query, got, ok, c.want)
		}
	}
}

func TestLookupCarriesLogoPath(t *testing.T) {
	r := Default()
	club, ok := r.Lookup("Real Madrid CF")
	if !ok {
		t.Fatal("Real Madrid CF must resolve")
	}
	if club.ClubID != "real_madrid" {
		t.Fatalf("club_id = %q", club.ClubID)
	}
	if club.LogoPath != "/clubs/real_madrid.png" {
		t.Fatalf("logo_path = %q", club.LogoPath)
	}
}

func TestUnknownIsNotFound(t *testing.T) {
	if _, ok := Default().Resolve("Some Unlisted FC"); ok {
		t.Error("unknown club must not resolve")
	}
}

func TestEveryClubHasLogoAndAliases(t *testing.T) {
	r := Default()
	all := r.All()
	if len(all) < 50 {
		t.Fatalf("registry too small: %d", len(all))
	}
	for _, c := range all {
		if c.LogoPath == "" || c.ClubID == "" || c.ShortName == "" {
			t.Errorf("incomplete club: %+v", c)
		}
		// self-resolution invariant
		if id, ok := r.Resolve(c.Name); !ok || id != c.ClubID {
			t.Errorf("name %q does not resolve to %q", c.Name, c.ClubID)
		}
	}
}

File: mobile/insight-azteca-flutter/test/club_registry_test.dart
// Club Registry (Azteca) — canonical resolution + bundled logo asset.
import 'package:flutter_test/flutter_test.dart';
import 'package:flutter/services.dart' show rootBundle;

import 'package:azteca/clubs/club_registry.dart';

void main() {
  TestWidgetsFlutterBinding.ensureInitialized();

  late ClubRegistry reg;

  setUpAll(() async {
    // Load the real bundled registry through the test asset bundle.
    final raw = await rootBundle.loadString('assets/clubs/logo_manifest.json');
    reg = ClubRegistry.instance..ingest(raw);
  });

  test('registry loads a substantial club set', () {
    expect(reg.isLoaded, isTrue);
    expect(reg.count, greaterThan(50));
    expect(reg.version, isNotEmpty);
  });

  test('canonical resolution: Man City variants → manchester_city', () {
    for (final q in const [
      'Manchester City',
      'Man City',
      'Manchester City FC',
      'manchester_city',
      'MCI',
    ]) {
      expect(reg.resolve(q), 'manchester_city', reason: 'query: $q');
    }
  });

  test('accent + punctuation folding', () {
    expect(reg.resolve('Atlético Mineiro'), 'atletico_mineiro');
    expect(reg.resolve('Flamengo'), 'flamengo');
    expect(reg.resolve('CR Flamengo'), 'flamengo');
  });

  test('lookup carries the bundled logo asset path', () {
    final club = reg.lookup('Real Madrid CF');
    expect(club, isNotNull);
    expect(club!.clubId, 'real_madrid');
    expect(club.logoAsset, 'assets/clubs/real/real_madrid.png');
  });

  test('unknown club resolves to null (badge falls back to initials)', () {
    expect(reg.resolve('Some Unlisted FC'), isNull);
    expect(reg.lookup(''), isNull);
  });
}
